### PR TITLE
Fix Go on Linux.

### PR DESCRIPTION
**Pull request by DoomHammer**
Pretty much what it says. Use Linux resources instead of Darwin ones.

### CODE REVIEW

File: Library/Formula/go.rb [hunk 1]
```diff
@@ -49,7 +52,7 @@ def install
 
     cd "src" do
       ENV["GOROOT_FINAL"] = libexec
-      ENV["GOOS"]         = "darwin"
+      ENV["GOOS"]         = OS.mac? ? "darwin" : "linux"
```
Comment (sjackman) [inline Library/Formula/go.rb]:
```ruby]
ENV["GOOS"]         = os = OS.mac? ? "darwin" : "linux"
```

File: Library/Formula/go.rb [hunk 2]
```diff
@@ -76,7 +79,11 @@ def install
         cd "src/golang.org/x/tools/cmd/vet/" do
           system "go", "build"
           # This is where Go puts vet natively; not in the bin.
-          (libexec/"pkg/tool/darwin_amd64/").install "vet"
+          if OS.mac?
+            (libexec/"pkg/tool/darwin_amd64/").install "vet"
+          else
+            (libexec/"pkg/tool/linux_amd64/").install "vet"
+          end
```
Comment (sjackman) [inline Library/Formula/go.rb]: Please use instead
```ruby
(libexec/"pkg/tool/#{os}_amd64/").install "vet"
```

File: Library/Formula/go.rb [hunk 3]
```diff
@@ -112,8 +119,13 @@ def caveats; <<-EOS.undent
...
     if build.with? "vet"
-      assert File.exist?(libexec/"pkg/tool/darwin_amd64/vet")
-      assert File.executable?(libexec/"pkg/tool/darwin_amd64/vet")
+      if OS.mac?
+        assert File.exist?(libexec/"pkg/tool/darwin_amd64/vet")
+        assert File.executable?(libexec/"pkg/tool/darwin_amd64/vet")
+      else
+        assert File.exist?(libexec/"pkg/tool/linux_amd64/vet")
+        assert File.executable?(libexec/"pkg/tool/linux_amd64/vet")
+      end
```
Comment (sjackman) [inline Library/Formula/go.rb]: Ditto, use `#{os}` as above.

**Comment by DoomHammer** [comment]
All suggested fixes in place.

File: Library/Formula/go.rb [hunk 4]
```diff
@@ -49,7 +52,7 @@ def install
 
     cd "src" do
       ENV["GOROOT_FINAL"] = libexec
-      ENV["GOOS"]         = "darwin"
+      ENV["GOOS"]         = os
```
Comment (sjackman) [inline Library/Formula/go.rb]: I think you're missing the following line?
```ruby
os = OS.mac? ? "darwin" : "linux"
```

**Comment by DoomHammer** [comment]
You are right. Fixed it.

**Comment by xu-cheng** [comment]
@DoomHammer Thanks for the pull request! 🎉

**Comment by sjackman** [comment]
Thanks for merging, Xu. I appreciate your help with Linuxbrew.

**Pull request closed without merging**